### ISSUE TITLE
Where did you derive number of pages?

### ISSUE THREAD

**Issue by vsoch** [penrose/arxiv-miner#4]
Hey @dormaayan ! You might be sleeping, but I have another question for you when you are back! I see that we have a number of pages file, and we read from there with csvreader:

```python
npagesFile = "npages.csv"
```
but this was derived outside of the script. Did you get this from the metadata via the arxiv API? E.g.: here is the `arxiv_comment`:

```python
{'affiliation': 'None',
 'arxiv_comment': '4 pages, 1 figure, presented at the 37th International Symposium on\n  Multiparticle Dynamics in Berkeley',
 'arxiv_primary_category': {'scheme': 'http://arxiv.org/schemas/atom',
  'term': 'nucl-th'},
```
Or did you derive it via some length metric of the actual LaTex (or something else)? I'm going to do the entire thing (on a per paper basis) in one schwoop so I won't have any summary csv file (and it wouldn't be feasible, given the number of papers!)

**Comment by dormaayan** [comment]
@vsoch Oh no, I don't sleep... Just finished another Skype with @wodeni and @joshsunshine  😂
`npages.csv` is actually a CSV with mappings from the paper ids to the number of pages for each one of them. I got this file from Paul

**Comment by vsoch** [comment]
It looks like it could be all of the papers - 

```bash
cat npages.csv | wc -l
1460335
```
I can definitely just use the file, but we should have a note (or a script?) that shows how the numbers were derived.

**Comment by dormaayan** [comment]
Agree, let's use it for now and see if we need more details on that.

**Comment by vsoch** [comment]
okay cool! A few more checks:

1. We are good with counting newlines in the line count?
2. Why did we count `thing` (the regular expression for beginning of figure twice, then subtract? This bit --> 

```python
#Given a fileName, return the number of figures in the file
def countFigures(fileContent):
    commented = fileContent.count("% "+ thing) + fileContent.count("%"+thing)
    return fileContent.count(thing) - commented
```

**Comment by dormaayan** [comment]
1. The new line is just a random metric I added, nothing informative since LaTex decide automatically on new lines and it is heavily dependent on the document layout.
2. This is just in order not to count all the commented figures (and also the one has a space), if you know any elegant way to do that with REs that would be great!

**Comment by vsoch** [comment]
So people submit papers with figures commented out? As in,  "I changed my mind" `%\begin{figure}` so it doesn't show up in the final publication? Geez. In that case, what you have looks good - I started diving into a regular expression with a look around but then decided...

![tenor](https://user-images.githubusercontent.com/814322/49587564-33f25280-f932-11e8-96f0-b278058ebee0.gif)

**Comment by dormaayan** [comment]
Yup

**Comment by dormaayan** [comment]
Hey @vsoch ,

I just had a look over the new statistics you produced, thanks!

I produced new statistics for avg figures per page (I think this is a stronger indication than avg figures per lines since lines doesn't tell us much)

However, I get this wierd graphs, which probably indicate information on the #pages in certain sub-subjects is still missing:

![unknown-2](https://user-images.githubusercontent.com/19957447/49795389-8670ac00-fd42-11e8-9643-499c82905bae.png)

![unknown](https://user-images.githubusercontent.com/19957447/49795513-d485af80-fd42-11e8-85b2-38ac3f413983.png)


Do you have any idea why it happens?

**Comment by dormaayan** [comment]
@vsoch, also, the results are weird. Does it mean that there are a lot of figures per page (more than 10 in CS!) which is unlikely...
Maybe we should recheck the code.

**Comment by vsoch** [comment]
yes this is what I mentioned re: missing data. We don't have all of it!

I just updated the notebook with work I did yesterday- in the future please don't just push to master (do a pull request first) because I didn't realize you would be working/change the notebook I was working on.

We don't have all of the data so the metrics that I show are for a subset. Take a look and let me know your thoughts!

PS it's like 6am here, no promises I'm not going back to sleep, haha.

**Comment by vsoch** [comment]
I mentioned that we need to take a catalog of all the data that there is, and then I can compare with what I have on Sherlock. Data was lost somewhere in the transfer and we need to know which files.

The violin plots show the distribution of figures across time for each topic, and then across all topics for each year. I was going to update the notebook to have figutres per page and an animated graphic, but I wasn't expecting you to ping so early. :)

**Comment by vsoch** [comment]
I'm adding some more notes (and better headers) now so will let you know when I update in a few minutes!

**Comment by vsoch** [comment]
@dormaayan I just updated the notebook, and added more notes / comments --> https://github.com/penrose/arxiv-miner/blob/master/analysis/arxiv-count-analysis.ipynb

The reason for the plots you generated is because we are missing data for articles for the number of pages. It artificially changes the metric, so to look at what we had, I filtered out those with missing (value not greater than 0) number of pages and the plots look more like I'd expect (no weird jumps!). I'll  again stress we need to do some inventory work, and maybe Josh has ideas for the best way to do this via the NAS?
